### PR TITLE
Cleanup resnet, implementation of inception block

### PR DESCRIPTION
**Pull request by zhanyinx**
## Proposed changes

Convert resnet from a combination of residual blocks and conv blocks to purely residual blocks. Implementation of inception block and the corresponding inception network. Fix zero division bug in rmse.

## Types of changes

 - [X] Bugfix (non-breaking change which fixes an issue)
 - [X] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
 - [ ] Documentation content update
 - [ ] Other (please describe):

## Checklist

Put an x in the boxes that apply. You can also fill these out after creating the PR.
If you're unsure about any of them, don't hesitate to ask. We're here to help!
This is simply a reminder of what we are going to look for before merging your code so nothing breaks.

 - [X] I have read the [contributing guidelines](https://deepblink.readthedocs.io/en/latest/contributing.html)
 - [X] Lint and unit tests pass locally with my changes (`tox`)
 - [X] I have added tests that prove my fix is effective or that my feature works
 - [X] I have added necessary documentation (if appropriate)
 - [X] Any dependent changes have been merged and published in downstream modules
 - [X] My commit messages format follow the project [git commit message format](https://github.com/slashsbin/styleguide-git-commit-message#commit-message-format)*
 - [ ] First time only - I have added myself / the copyright holder to the authors.rst file

💔 Thank you!

\* Please note we only use the following (same semantic meaning as described) 🎨📰📝🐎📚🐛🚑🔥🚜🔨☔💚💎✨⬆️⬇️

### CODE REVIEW

File: deepblink/networks/_networks.py [hunk 1]
```diff
@@ -70,13 +114,16 @@ def upconv_block(
...
         filters: Number of convolutional filters applied.
         n_convs: Number of convolution+relu blocks after concatenation.
+        dropout: If > 0, a dropout layer will be added.
     """
     x = inputs
     x = tf.keras.layers.UpSampling2D()(x)
     x = tf.keras.layers.Conv2D(filters=filters, **OPTIONS_CONV)(x)
     x = tf.keras.layers.Activation("relu")(x)
+    if dropout > 0:
+        x = tf.keras.layers.Dropout(dropout)(x)
```
Comment (BBQuercus) [inline deepblink/networks/_networks.py:124]: You could remove this if statement as dropout with `0.0` is equal to no layer.

File: deepblink/networks/fcn.py [hunk 2]
```diff
@@ -11,27 +11,31 @@
...
 ) -> tf.keras.models.Model:
     """Simplest FCN architecture without skips or dropout.
 
     Arguments:
-        dropout: Percentage of dropout only for resnet architecture.
+        dropout: Percentage of dropout after each conv layer.
         cell_size: Size of one cell in the prediction matrix.
         filters: Log2 number of filters in the first convolution layers.
         n_convs: Number of convolution layers in each convolution block.
+        n_extra_down: extra downsampling followed by same number of up sampling.
```
Comment (BBQuercus) [inline deepblink/networks/fcn.py:27]: For a later commit probably change `n_extra_down`. Slightly strange name but don't have a better one in mind just yet.

File: deepblink/networks/fcn.py [hunk 3]
```diff
@@ -43,20 +47,31 @@ def fcn(
     # If cell_size is 1, we need to upsample 2 == int(2 // cell_size)
     # If cell_size is 2, we need to upsample 1 == int(2 // 2)
     n_up = int(2 // cell_size)
-    for n in range(n_up):
+    print(n_up + n_extra_down)
```
Comment (BBQuercus) [inline deepblink/networks/fcn.py:50]: Remove this print statement.

File: deepblink/networks/resnet.py [hunk 4]
```diff
@@ -35,15 +29,9 @@ def resnet(
...
-        x = conv_block(
-            inputs=x, filters=2 ** (filters + n), n_convs=n_convs, dropout=dropout
-        )
+    for n in range(n_down + n_extra_down):
         x = residual_block(inputs=x, filters=2 ** (filters + n))
-        if conv_after_res:
-            x = conv_block(
-                inputs=x, filters=2 ** (filters + n), n_convs=n_convs, dropout=dropout,
-            )
+        if dropout:
```
Comment (BBQuercus) [inline deepblink/networks/resnet.py:34]: Same thing here

File: deepblink/networks/resnet.py [hunk 5]
```diff
@@ -53,31 +41,17 @@ def resnet(
...
+    for n in range(n_up + n_extra_down):
         x = residual_block(inputs=x, filters=2 ** (filters + 1 - n))
-        if conv_after_res:
-            x = conv_block(
-                inputs=x,
-                filters=2 ** (filters + 1 - n),
-                n_convs=n_convs,
-                dropout=dropout,
-            )
+        if dropout:
```
Comment (BBQuercus) [inline deepblink/networks/resnet.py:46]: and here

File: deepblink/networks/inception.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
+# pylint: skip-file
```
Comment (BBQuercus) [inline deepblink/networks/inception.py:1]: Looks complicated but is fine. If FCN worked better than this (also keep n_params in mind) then there's no need to invest too much time here.

**Review by BBQuercus — CHANGES_REQUESTED** [review]
Some comments throughout the files. Only things to change are `if dropout` and print statement.

**Review by BBQuercus — APPROVED** [review]
